src/Moat.hs

```diff
@@ -1027,13 +1057,30 @@ getFreeTyVar = \case
   _ -> Nothing
 
 -- make a struct field pretty
-prettyField :: Options -> Name -> Type -> Exp
-prettyField Options {..} name ty =
-  tupE
-    [ stringE (onHeadWith lowerFirstField (fieldLabelModifier (nameStr name))),
-      toMoatTypeEPoly ty
+prettyField :: Options -> Name -> Type -> Maybe String -> Exp
+prettyField Options {..} name ty doc =
+  RecConE
+    'Field
+    [ ('fieldName, stringE (onHeadWith lowerFirstField (fieldLabelModifier (nameStr name)))),
+      ('fieldType, toMoatTypeEPoly ty),
+      ('fieldDoc, prettyDoc doc)
     ]
 
+-- prettify doc string as an Exp
+prettyDoc :: Maybe String -> Exp
+prettyDoc = \case
+  Nothing -> ConE 'Nothing
+  Just doc -> AppE (ConE 'Just) (stringE doc)
+
+-- get doc string for Name
+getDoc :: Name -> Q (Maybe String)
+
+#if MIN_VERSION_template_haskell(2,18,0)
```

Comment:
Do we also need to do this in the import for CI?

Reply:
I think @parsonsmatt has the solution, which is to not shadow `getDoc` and give the function a new name.